code/modules/projectiles/guns/energy/laser.dm

```diff
@@ -138,3 +164,69 @@
 
 /obj/item/gun/energy/laser/redtag/hitscan
 	ammo_type = list(/obj/item/ammo_casing/energy/laser/redtag/hitscan)
+
+/obj/item/external_lens
+	name = "external lens: scatter ray"
+	icon_state = "external_scatter"
+	desc = "external lens."
+	icon = 'icons/obj/guns/energy.dmi'
+	var/stored_ammo_type = /obj/item/ammo_casing/energy/laser/scatter
+
+/obj/item/external_lens/BTC
+	name = "external lens: ticket dispenser"
+	stored_ammo_type = /obj/item/ammo_casing/energy/laser/bitcoin
+	icon_state = "external_bitcoin"
+
+/obj/item/external_lens/RIC
+	name = "external lens: bouncing ray"
+	stored_ammo_type = /obj/item/ammo_casing/energy/laser/rico
+
+/obj/item/external_lens/TRA
+	name = "external lens: tracer ray"
+	stored_ammo_type = /obj/item/ammo_casing/energy/laser/tracer
+
+/obj/item/external_lens/SHK
+	name = "external lens: shocking ray"
+	stored_ammo_type = /obj/item/ammo_casing/energy/laser/shock
+
+/obj/item/external_lens/BLN
+	name = "external lens: blinding ray"
+	stored_ammo_type = /obj/item/ammo_casing/energy/laser/blinding
+
+/obj/item/external_lens/STH
+	name = "external lens: stealth ray"
+	stored_ammo_type = /obj/item/ammo_casing/energy/laser/invisible
+	icon_state = "external_stealth"
+
+/obj/item/external_lens/INC
+	name = "external lens: incendiary ray"
+	stored_ammo_type = /obj/item/ammo_casing/energy/laser/incendiary
+
+/obj/item/external_lens/HEA
+	name = "external lens: heavy plasma bolt"
+	stored_ammo_type = /obj/item/ammo_casing/energy/laser/heavy
+	icon_state = "external_heavy"
+
+/obj/item/external_lens/ECO
+	name = "external lens: low power consuption ray"
+	stored_ammo_type = /obj/item/ammo_casing/energy/laser/lowenergy
+
+/obj/item/external_lens/SYP
+	name = "external lens: cell syphoning"
+	stored_ammo_type = /obj/item/ammo_casing/energy/laser/syphon
+
+/obj/item/external_lens/SYP/Initialize()
+	. = ..()
+	AddComponent(/datum/component/Scharge)
+
+/datum/component/Scharge/Initialize()
+	. = ..()
+	RegisterSignal(parent,"syphon", .proc/charge)
+
+/datum/component/Scharge/proc/charge(mob/FH) //doenst work

```

Comment:
The first argument is always the signal source as documented in the signals defines file.